assets/app/scripts/directives/fromFile.js

```diff
@@ -0,0 +1,316 @@
+        }
+
+        function checkResource(item) {
+          // Check if the resource already exists. If it does, replace it spec with the new one.
+          return DataService.get(APIService.kindToResource(item.kind), item.metadata.name, $scope.context, {errorNotification: false}).then(
+            // resource does exist
+            function(resource) {
+              resource.spec = item.spec;
+              $scope.updateResources.push(resource);
+            },
+            // resource doesn't exist
+            function() {
+              $scope.createResources.push(item);
+          });
+        }
+
+        function createResourceList(){
+          var titles = {
+            started: "Creating in project " + $scope.projectName,
+            success: "Created in project " + $scope.projectName,
+            failure: "Failed to create in project " + $scope.projectName
+          };
+          var helpLinks = {};
+          TaskList.add(titles, helpLinks, function() {
+            var d = $q.defer();
+
+            DataService.listAction($scope.createResources, $scope.context).then(
+              function(result) {
+                var alerts = [];
+                var hasErrors = false;
+                if (result.failure.length > 0) {
+                  hasErrors = true;
+                  result.failure.forEach(
+                    function(failure) {
+                      alerts.push({
+                        type: "error",
+                        message: "Cannot create " + humanize(failure.object.kind).toLowerCase() + " \"" + failure.object.metadata.name + "\". ",
+                        details: failure.data.message
+                      });
+                    }
+                  );
+                  result.success.forEach(
+                    function(success) {
+                      alerts.push({
+                        type: "success",
+                        message: "Created " + humanize(success.kind).toLowerCase() + " \"" + success.metadata.name + "\" successfully. "
+                      });
+                    }
+                  );
+                } else {
+                  var alertMsg;
+                  if ($scope.isList) {
+                    alertMsg = "All items in list were created successfully.";
+                  } else {
+                    alertMsg = $scope.resourceKind + " " + $scope.resourceName + " was successfully created.";
+                  }
+                  alerts.push({ type: "success", message: alertMsg});
+                }
+                d.resolve({alerts: alerts, hasErrors: hasErrors});
+              }
+            );
+            return d.promise;
+          });
+        }
+
+
+        function updateResourceList(){
+          var titles = {
+            started: "Updating in project " + $scope.projectName,
+            success: "Updated in project " + $scope.projectName,
+            failure: "Failed to update in project " + $scope.projectName
```

Comment:
"Updating resources in project..."
"Updated resources in project..."
"Failed to update resources..."